core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -1007,24 +1007,31 @@ private CompiledQuery alterTableDropDetachOrAttachPartitionByList(
                 }
                 break;
             }
-            if (Chars.equals(tok, ',')) {
+            if (Chars.equals(tok, ',') || Chars.equals(tok, ';')) {
                 throw SqlException.$(lexer.lastTokenPosition(), "partition name missing");
             }
-            final CharSequence partitionName = GenericLexer.unquote(tok);
+            final CharSequence partitionName = GenericLexer.unquote(tok); // potentially a full timestamp, or part of it
             final int partitionNamePosition = lexer.lastTokenPosition();
 
             // reader == null means it's compilation for WAL table
             // before applying to WAL writer
             if (reader != null) {
-                final long timestamp;
                 try {
-                    timestamp = PartitionBy.parsePartitionDirName(partitionName, reader.getPartitionedBy());
+                    // rtrim partition name to appropriate size
+                    int partitionBy = reader.getPartitionedBy();
+                    int len = PartitionBy.getPartitionDirNameRootLen(partitionBy);
+                    if (len == -1) {
+                        len = partitionName.length();
+                    } else if (partitionName.length() < len) {
+                        throw SqlException.position(lexer.lastTokenPosition())
+                                .put("timestamp has too low resolution to determine partition [ts=").put(partitionName).put(']');
+                    }
+                    long timestamp = PartitionBy.parsePartitionDirName(partitionName, 0, len, partitionBy);
```

Comment:
I'm not sure if that's a valid approach for ISO week. The thing is that ISO week is different from the calendar week, so it's not possible to extract it as a substring from the given timestamp string.

Also, asking the user to properly format the timestamp is inconvenient. We may want to allow users to use bind variables in the partition list so that they can use timestamp type suitable for their programming language. Say, in Java you may fetch the min timestamp as a `Date` and then provide this object as a bind variable value without having to think of ts formatting at all.

Reply:
Addressed in commit 4faedc005a3384402cd885e3b56fdd952ed80921